test/docfx.Test/e2e/E2ETest.cs

```diff
@@ -154,21 +164,28 @@ private static void DeleteDirectory(string targetDir)
             Directory.Delete(targetDir, false);
         }
 
-        private static IEnumerable<string> FindTestSpecHeadersInFile(string path)
+        private static IEnumerable<(string, bool)> FindTestSpecHeadersInFile(string path)
         {
-            var sections = File.ReadAllText(path).Split("\n---", StringSplitOptions.RemoveEmptyEntries);
+            var content = File.ReadAllText(path);
+            var sections = content.Split("\n---", StringSplitOptions.RemoveEmptyEntries);
 
             foreach (var section in sections)
             {
+                var only = false;
                 var yaml = section.Trim('\r', '\n', '-');
                 var header = YamlUtility.ReadHeader(yaml) ?? "";
 
                 foreach (var ch in Path.GetInvalidPathChars())
                 {
                     header = header.Replace(ch, ' ');
                 }
-
-                yield return header.Replace('/', ' ').Replace('\\', ' ');
+#if DEBUG
+                if (yaml.Contains("only: true"))
```

Comment:
How about putting the flag just in the header? Like [only] in the beginning to avoid going through the entire YAML content